src/ethereum/frontier/vm/instructions/memory.py

```diff
@@ -0,0 +1,141 @@
+"""
+Ethereum Virtual Machine (EVM) Memory Instructions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. contents:: Table of Contents
+    :backlinks: none
+    :local:
+
+Introduction
+------------
+
+Implementations of the EVM Memory instructions.
+"""
+
+from ethereum.base_types import U256
+from ethereum.frontier.vm.memory import memory_read_bytes, memory_write
+
+from .. import Evm
+from ..gas import (
+    GAS_BASE,
+    GAS_VERY_LOW,
+    calculate_gas_for_extending_memory,
+    subtract_gas,
+)
+from ..memory import memory_extend
+from ..stack import pop, push
+
+
+def mstore(evm: Evm) -> None:
+    """
+    Stores a word to memory.
+
+    Parameters
+    ----------
+    evm :
+        The current EVM frame.
+
+    Raises
+    ------
+    StackUnderflowError
+        If `len(stack)` is less than `2`.
+    OutOfGasError
+        If `evm.gas_left` is less than
+        `GAS_LOW` + gas needed to extend memeory.
+    """
+    start_position = pop(evm.stack)
+    value = pop(evm.stack).to_be_bytes32()
+    gas_extend_memory = calculate_gas_for_extending_memory(
+        evm.memory, start_position, U256(1)
```

Comment:
Not saying that anything is incorrect here, but `U256(1)` jumps out as odd compared to the following two `U256(32)`. If this is correct, maybe add a short comment describing what you're doing?

Reply:
maybe I should clarify in `calculate_gas_for_extending_memory` that the size parameter is the number of **bytes** by which we want to extend the memory. I can add comment here too.

Reply:
So here we check the price for extending by one byte, and if it costs anything to add one byte, we change the cost to extend by 32 bytes?

Reply:
Yeah, that's because the memory is a word addressable array. That logic is handled by the `ceil32` method. 
not sure If I understood your question completely and answered it.